### PR TITLE
RFR: Use delete_by_query to delete models in purge tool

### PR DESCRIPTION
**Pull request by lakshmi-kannan**

### CODE REVIEW

File: st2common/st2common/persistence/trigger.py [hunk 1]
```diff
@@ -84,3 +84,7 @@ class TriggerInstance(Access):
     @classmethod
     def _get_impl(cls):
         return cls.impl
+
+    @classmethod
+    def delete_by_query(cls, **query):
```
Comment (Kami) [inline st2common/st2common/persistence/trigger.py]: Can't we just add this to the base class (`Access`)?

Comment (lakshmi-kannan) [inline st2common/st2common/persistence/trigger.py]: This is done so we have control over which models should support purge. It's a little confusing for me when I looked at the code.

**Comment by Kami** [comment]
Do we actually install those tools somewhere or the user needs to clone the repo to use them?

I quickly checked st2common/bin/packaging and it doesn't seem so.

**Comment by lakshmi-kannan** [comment]
@Kami yep, we want to test this out in st2build before making it available as tool. I stress tested in my VM some. build and ops box testing would be better.

**Comment by Kami** [comment]
Cool, sounds good.

File: st2common/st2common/persistence/trigger.py (hunk 1 above)
Comment (Kami) [inline st2common/st2common/persistence/trigger.py]: Yeah, assumed something like that was the case.

**Pull request merged**